spec/apis/check-activity-functions.spec.js

```diff
@@ -45,76 +45,92 @@ const dummyApp3 = {
   },
 };
 
-const activityFunctions = {
-  1: activityFunction1,
-  2: activityFunction2,
-  3: activityFunction3,
+const dummyApp4 = {
+  bootstrap() {
+    return Promise.resolve();
+  },
+  mount() {
+    return Promise.resolve();
+  },
+  unmount() {
+    return Promise.resolve();
+  },
+  unload() {
+    return Promise.resolve();
+  },
 };
 
 describe(`checkActivityFunctionsApi`, () => {
   beforeAll(() => {
     singleSpa.registerApplication(
       "test1",
       () => Promise.resolve(dummyApp1),
-      activityFunctions[1]
+      (location) => location.href.indexOf("one") !== -1
     );
     singleSpa.registerApplication(
       "test2",
       () => Promise.resolve(dummyApp2),
-      activityFunctions[2]
+      (location) => location.href.indexOf("two") !== -1
     );
     singleSpa.registerApplication({
       name: "test3",
       app: Promise.resolve(dummyApp3),
-      activeWhen: activityFunctions[3],
+      activeWhen: (location) => location.href.indexOf("three") !== -1,
+    });
+    singleSpa.registerApplication({
+      name: "test4",
+      app: Promise.resolve(dummyApp4),
```

Comment:
`app` should be either an object or a function that resolves with an object. It should not be a promise. We should change this test and also the dummyApp3 test. And should fix registerApplication to enforce this.